### ISSUE TITLE
Incorrect dynamic parameter type changes crash the node

### ISSUE THREAD

**Issue by stonier** [stonier/ueye_cam#26]
Currently if you try to send it an incorrect type, it will just bomb. Should instead handle and reject the change.

```
$ ros2 param set ueye_cam blue_gain 0

terminate called after throwing an instance of 'rclcpp::ParameterTypeException'
[ueye-1]   what():  expected [integer] got [bool]
[ERROR] [ueye-1]: process has died [pid 30068, exit code -6, cmd 'stdbuf -o L /home/motionsim/ueye_camera_testing/lib/ueye_cam/standalone_node __params:=/tmp/launch_params__6s0cb_7'].
```

**Comment by stonier** [comment]
Need to a) catch the exception and handle it, or b) check the parameter's type (example [demo](https://github.com/ros2/demos/blob/master/demo_nodes_cpp/src/parameters/even_parameters_node.cpp)).

**Comment by stonier** [comment]
* Relevant code in `Node::onParameterChange`

**Comment by stonier** [comment]
Fixed in #32.
